semantic_router/index/pinecone.py

```diff
@@ -50,14 +50,10 @@ class PineconeIndex(BaseIndex):
 
     def __init__(self, **data):
         super().__init__(**data)
-        self._initialize_client()
-
-        self.type = "pinecone"
-        self.client = self._initialize_client()
-        if not self.index_name.startswith(self.index_prefix):
-            self.index_name = f"{self.index_prefix}{self.index_name}"
-        # Create or connect to an existing Pinecone index
-        self.index = self._init_index()
+        clean_route = clean_route_name(self.route)
+        # Use SHA-256 for a more secure hash
+        utterance_id = hashlib.sha256(self.utterance.encode()).hexdigest()
+        self.id = f"{clean_route}#{utterance_id}"
```

Comment:
This seems wrong? We are no longer initialising the index and are creating a self.id attribute?